### ISSUE TITLE
Radio Button text shows true/false if the folder path has a space in it

### ISSUE THREAD

**Issue by pingouyn** [starcitizen-lug/lug-helper#36]
Ubuntu 21.10 - Impish.  Up to date.
Plasma 5.22.5
KDE framework 5.86.0
Qt 5.15.2
Kernel 5.15.35-xanmod1 (64-bit)
Graphics X11
File manager: dolphin
Language: french

lug-helper-3.17-shaders (or lug-helper-master, same)

If the folder path in which the lug-helper is located has a space, the radio buttons on the main page menu of the lug-helper show 'true' on the first line and 'false' on all the others.  It is impossible to perform any action either, the lug-helper just exists regardless of what is pressed.

If the folder path doesn't have any space the lug-helper works flawlessly.

I tried different path depths, partitions, same results everywhere.
Davy

**Comment by Termuellinator** [comment]
I just tried to replicate this by renaming the folder to "lug helper", but everything seems fine. Do you use an uncommon shell by any chance? I'm using zsh, bash should be used by some members for sure and i think there are also some that mentioned fish, so those should be fine i'd expect.

**Comment by pingouyn** [comment]
ok, it's my 3rd month on Linux (just evenings for now) so had to check online what a shell was...  Sorry for that.
My ubuntu is using bash which I believe is the default.
Also I open the lug-helper using the GUI dolphin, not from the command line (not sure if the shell is used or not when I'm under dolphin, but if I understand what I read it is).

This location works fine for the helper:
/home/davy/lug-helper-3.17-shaders/
This one doesn't:
/home/davy/lug helper-3.17-shaders/
(screenshot attached)
![Screenshot_20220424_223740](https://user-images.githubusercontent.com/87666912/164995647-f058b9ca-4de3-4c8d-9e6e-b37180e7b1f8.png)

**Comment by Termuellinator** [comment]
that is really really odd - the only way this could come up would be if the zen_options array is shifted by one :thinking: 
I wouldn't put it above ubuntu to do such a thing, but i found nothing of the like (and that should probably yield a major shitstorm tbh), so i kind of doubt that's the case...
I'll have to see when i find the time to fire up an ubuntu vm to check if that's the culprit. 
The only other thing i could imagine is some really strange language/locale issue - would it be possible for you to check if the issue also occurs when you switch your system to english?

**Comment by pingouyn** [comment]
Sure, anything I can do to help.

I changed the language to american english everywhere I could find (for system language, spell checks and formats), relogged, same result.
I changed the compositor to Wayland from x11 just in case, relogged, same.
The lug-helper is saved under lvm volumes, but I tried on a simple Windows ntfs partition (outside the linux lvm), on a Windows RAID-1 ntfs partition (outside the linux lvm), on a storage space "raid 1" windows drive, same results for all attemps.

I'll try to reboot with the default kernel to see if it makes a difference.
brb

I can try other stuffs if you can think of anything.
D

**Comment by pingouyn** [comment]
ok I found something interesting.  From there I believe you should be able to debug the script.

In a QEMU-KVM, I installed a fresh Ubuntu (gnome this time) version 22.04.
System up-to-date.
I installed Zenity from the software manager.
Downloaded the latest "lug-helper-3.17-shaders" in /home/downloads/.

This folder name works: "lug-helper-3.17-shaders"  (I can see a terminal opening up then the helper itself).
This folder name doesn't work: "lug -helper-3.17-shaders" (space added after "lug".  I can see the terminal but it quickly closes and the helper never opens up).

Hope that helps!

**Comment by Termuellinator** [comment]
we only have `helper_dir` refering to the location of the helper script and that one is quoted.  I also can't see any other thing that could be causing this. Everything that is called up to that point is self-contained. As the TRUE/FALSE entries are correct for the default selection, it seems everything up to line 350 (in this branch) is executed correctly and the error is somewhere in line 352 - i just can't see how. It could be a bug in this zenity-version, but if i downgrade to 3.32.0, i still can't reproduce it. Hopefully i'll manage to take a look at it in a VM tomorrow.

@pingouyn if you want, you could insert `printf "zen_options is: ${zen_options[*]} \n menu_options is: ${menu_options[*]}"` at line 351 in this branch and post the terminal-output when calling ./lug-helper.sh here to see if the array is shifted for you at this stage.

**Comment by pingouyn** [comment]
I had to reinstall that release of Linux after a lvm subvolume resizing that went bad (still learning about linux before jumping on it full time).  Now trying btrfs, I know...
I'll check if the error still persists with Kubuntu 22.04 and if so I'll continue to debug this as per both your suggestions above.  I need a few days... D

**Comment by the-sane** [comment]
>  it seems everything up to line 350 (in this branch) is executed correctly and the error is somewhere in line 352 - i just can't see how. 

The variable $lug_logo in that line is unquoted, and it contains $helper_dir within it from the top of the script. That means it could be expanded into two arguments and would potentially mess everything up depending on the shell and how zenity parses those arguments. That looks like a good candidate for the problem to me.

I’m not sure if there was a reason why we had it unquoted?

**Comment by Termuellinator** [comment]
@the-sane man, sometimes i really think i'm blind - good catch!
It really seems to be a likely candidate - if i put a space in my path AND rename my lug_logo.png in /usr/share/pixmaps i at least don't get a zenity-window at all. So that would explain the difference, too.

@pingouyn i've quoted the path and for me the behaviour seems normal with that, even without the logo in pixmaps - could you give that a try?

**Comment by pingouyn** [comment]
ok, I'm back, sorry for the late reply - was stuck 2 days in Win$ whilst rebuilding a raid 1 dynamic disk.. pfff).  Turns out dynamic disk are the old way of doing "storage spaces"... rofl
I'm also now on kubuntu 22.04.  I couldn't run the helper but soon realised I was missing 'curl'.
Anyways, the issue is still the same in kubuntu 22.04 with zenity installed by default and curl added.

@the-sane Yes same issue with the helper release dated 18-march from the link you pasted.

@Termuellinator I've added the line "printf..." at 351, here's the log when run from the CLI:
zen_options is: TRUE Preflight Check (System Optimization) FALSE Install Star Citizen FALSE Deploy Easy Anti-Cheat Workaround FALSE Manage Lutris Runners FALSE Manage Lutris DXVK Versions FALSE Maintenance and Troubleshooting FALSE Get a random Penguin's Star Citizen referral code 
 menu_options is: Preflight Check (System Optimization) Install Star Citizen Deploy Easy Anti-Cheat Workaround Manage Lutris Runners Manage Lutris DXVK Versions Maintenance and Troubleshooting Get a random Penguin's Star Citizen referral code Quit

It doesn't look shifted to me if I'm not wrong.

@Termuellinator I now quoted $lug_logo (to "$lug_logo") on line 351 of the 18-march release of the helper, and the interface displays CORRECTLY.  Tried with and without, the solve is repeatable.

The issue is now solved!  You guys form a great team!
Thank you so much for your time developing the helper and helping so many of us install and use SC on linux the easy way.
